src/agent0/hyperfuzz/system_fuzz/run_fuzz_bots.py

```diff
@@ -7,12 +7,113 @@
+
+# The initial time stretch APR
+INITIAL_TIME_STRETCH_APR_RANGE: tuple[float, float] = (0.005, 0.5)
+# The variable rate to set after each episode
+VARIABLE_RATE_RANGE: tuple[float, float] = (0, 1)
+# How much to advance time between episodes
+ADVANCE_TIME_SECONDS_RANGE: tuple[int, int] = (0, ONE_DAY_IN_SECONDS)
+# The fee percentage. The range controls all 4 fees
+FEE_RANGE: tuple[float, float] = (0.0001, 0.2)
+
+
+def generate_fuzz_hyperdrive_config(rng: Generator, log_to_rollbar: bool, rng_seed: int) -> ILocalHyperdrive.Config:
+    """Fuzz over hyperdrive config.
+
+    Arguments
+    ---------
+    rng: np.random.Generator
+        Random number generator.
+    log_to_rollbar: bool
+        If True, log errors to rollbar.
+    rng_seed: int
+        Seed for the rng.
+
+    Returns
+    -------
+    ILocalHyperdrive.Config
+        Fuzzed hyperdrive config.
+    """
+    # Position duration must be a multiple of checkpoint duration
+    # To do this, we calculate the number of checkpoints per position
+    # and adjust the position duration accordingly.
+    position_duration_hours = int(rng.integers(POSITION_DURATION_HOURS_RANGE[0], POSITION_DURATION_HOURS_RANGE[1]))
+    checkpoint_duration_hours = int(
+        rng.integers(CHECKPOINT_DURATION_HOURS_RANGE[0], CHECKPOINT_DURATION_HOURS_RANGE[1])
+    )
+
+    # Checkpoint duration must be a multiple of `factory_checkpoint_duration_resolution`
+    checkpoints_per_position_duration = position_duration_hours // checkpoint_duration_hours
+    position_duration_hours = checkpoint_duration_hours * checkpoints_per_position_duration
+    # There's a chance the new position duration was truncated to be less than the minimum
+    # If that's the case, we use the ceil instead.
+    if position_duration_hours < POSITION_DURATION_HOURS_RANGE[0]:
+        position_duration_hours = checkpoint_duration_hours * (checkpoints_per_position_duration + 1)
+
+    # Sanity check
+    assert POSITION_DURATION_HOURS_RANGE[0] <= position_duration_hours <= POSITION_DURATION_HOURS_RANGE[1]
+
+    # Convert checkpoint duration and position duration to seconds
+    position_duration = position_duration_hours * ONE_HOUR_IN_SECONDS
+    checkpoint_duration = checkpoint_duration_hours * ONE_HOUR_IN_SECONDS
+
+    initial_time_stretch_apr = FixedPoint(
+        rng.uniform(INITIAL_TIME_STRETCH_APR_RANGE[0], INITIAL_TIME_STRETCH_APR_RANGE[1])
+    )
+    # Generate flat fee in terms of APR
+    flat_fee = FixedPoint(rng.uniform(FEE_RANGE[0], FEE_RANGE[1]) * (position_duration / ONE_YEAR_IN_SECONDS))
+
+    return ILocalHyperdrive.Config(
+        preview_before_trade=True,
+        rng=rng,
+        log_to_rollbar=log_to_rollbar,
+        rollbar_log_prefix="localfuzzbots",
+        crash_log_level=logging.CRITICAL,
+        crash_report_additional_info={"rng_seed": rng_seed},
+        # Initial hyperdrive config
+        initial_liquidity=FixedPoint(rng.uniform(INITIAL_LIQUIDITY_RANGE[0], INITIAL_LIQUIDITY_RANGE[1])),
```

Comment:
might be cleaner to use splats here; `rng.uniform(*INITIAL_LIQUIDITY_RANGE)` would work right?